aptos-move/e2e-move-tests/src/tests/resource_groups.rs

```diff
@@ -341,3 +430,40 @@ fn verify_resource_group_upgrades() {
     let result = h.publish_package(&account, path.path());
     assert_vm_status!(result, StatusCode::CONSTRAINT_NOT_SATISFIED);
 }
+
+#[test]
+fn verify_unsafe_resource_group_upgrades() {
+    let mut h = MoveHarness::new_with_features(vec![], vec![FeatureFlag::SAFER_RESOURCE_GROUPS]);
+    let account = h.new_account_at(AccountAddress::from_hex_literal("0xf00d").unwrap());
+
+    // Initial code
+    let source = r#"
+        module 0xf00d::M {
+            #[resource_group(scope = address)]
+            struct ResourceGroup { }
+
+            struct NotResourceGroup { }
+        }
+        "#;
+    let mut builder = PackageBuilder::new("Package");
+    builder.add_source("m.move", source);
+    let path = builder.write_to_temp().unwrap();
+    let result = h.publish_package(&account, path.path());
+    assert_success!(result);
+
+    // Incompatible promotion of ResourceGroup
+    let source = r#"
+        module 0xf00d::M {
+            #[resource_group(scope = address)]
+            struct ResourceGroup { }
+
+            #[resource_group(scope = address)]
+            struct NotResourceGroup { }
+        }
+        "#;
+    let mut builder = PackageBuilder::new("Package");
+    builder.add_source("m.move", source);
+    let path = builder.write_to_temp().unwrap();
+    let result = h.publish_package(&account, path.path());
+    assert_success!(result);
```

Comment:
would be nice to comment it's going through because safety check is disabled.

Reply:
look at the name of the test

Reply:
while that might be true, I think having both the test name and the first line of the test show that we are disabling a feature should be indicative enough for anyone that wants to understand why that test might all of a sudden fail. If we need comments for this, then we have failed, we -- I have failed so hard.